core/src/test/scala/integration/kafka/api/EndToEndAuthorizationTest.scala

```diff
@@ -170,7 +171,16 @@ trait EndToEndAuthorizationTest extends IntegrationTestHarness with SaslSetup {
     * Tests the ability of producing and consuming with the appropriate ACLs set.
     */
   @Test
-  def testProduceConsume {
+  def testProduceConsumeViaAssign {
+    testProduceConsume(assign = true)
+  }
+
+  @Test
+  def testProduceConsumeViaSubscribe {
+    testProduceConsume(assign = false)
+  }
+
+  private def testProduceConsume(assign : Boolean) {
```

Comment:
Instead of this, I think I would have a method for setting ACLs and waiting for them to be propagated. And then each `testProduceConsumerVia*` method can just compose the calls together instead of having this boolean that is a bit weird.